### PR TITLE
single config specifications of both db and tasks and multi-datastores

### PR DESCRIPTION
**Pull request by mvcrouse**
…and adding multi datastores

## Related Issue
Supports #ISSUE_NUMBER

## Related PRs
This PR is not dependent on any other PR

## What this PR does / why we need it

This PR adds the ability to have multiple data stores as well as adds the ability to use a single config to specify all jobs

## If applicable**
- [ ] this PR contains documentation
- [X] this PR contains unit tests
- [X] this PR has been tested for backwards compatibility

### CODE REVIEW

File: fms_dgt/__main__.py [hunk 1]
```diff
@@ -37,17 +36,15 @@ def add_base_args(parser: argparse.ArgumentParser):
...
         metavar="CONFIG_PATH",
-        help="Additional path to include if there are overrides for data builder config files.",
+        help="Path that specifies both data builder configs and tasks.",
     )
     group.add_argument(
         "--data-paths",
         type=str,
-        nargs="+",
-        default=DEFAULT_DATA_PATH,
+        nargs="*",
```
Comment (sivasankalpp) [inline fms_dgt/__main__.py:47]: Does this mean that `--data-paths` is now an optional field? What happens if we pass both `-config-path` and `--data-paths`?

File: fms_dgt/generate_data.py [hunk 2]
```diff
@@ -3,24 +3,38 @@
...
+
+    if config_path:
+        data_paths, config_overrides = utils.load_joint_config(config_path)
+    elif data_paths:
+        data_paths, config_overrides = data_paths, None
+    else:
+        raise ValueError(
+            f"One of ['data-paths', 'config-path'] must be provided in the arguments"
+        )
+
```
Comment (sivasankalpp) [inline fms_dgt/generate_data.py:37]: Ah this answers my previous question lol

**Review by sivasankalpp — APPROVED** [review]
Great change! Just a few questions that I've left above. 
We also need to update the README and design doc with these changes (either in this PR or I could take a crack at it in the next one).

File: configs/all_api_tasks.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+        type: genai
+        temperature: 0.5
+        model_id_or_path: mistralai/mixtral-8x7b-instruct-v01
+      - name: val2
+        filter: true
+        threshold: 0.9
+tasks:
+  - data/generation/code/apis/glaive/detection/multiple_apis/qna.yaml
+  - data/generation/code/apis/glaive/detection/single_api/qna.yaml
+  - data/generation/code/apis/glaive/detection/yes_no/qna.yaml
```
[the remark this answers is not in the document]
Comment (mvcrouse) [reply configs/all_api_tasks.yaml:14]: Yep, only will override the specified databuilders and has special handling for overriding the settings of specific blocks

[the remark this answers is not in the document]
Comment (sivasankalpp) [reply configs/all_api_tasks.yaml:14]: Makes sense!

**Comment by mvcrouse** [comment]
> Great change! Just a few questions that I've left above. We also need to update the README and design doc with these changes (either in this PR or I could take a crack at it in the next one).

Yes, good catch, I'll do a deep run through in the next PR to make sure everything is updated

**Pull request merged**